plugins/providers/hyperv/action/configure.rb

```diff
@@ -1,69 +1,128 @@
           switches = env[:machine].provider.driver.execute(:get_switches)
           if switches.empty?
             raise Errors::NoSwitches
           end
 
-          switch = nil
+          # Use the hyper-v Default Switch for NAT if a bridged switch wasn't found
+          default_switch = managed_switches.find{ |s|
+            s["SwitchType"].downcase == "nat"
+          }
+
+          # Attach explicitly defined networks starting at number 2 because 1 is the NAT switch
+          additional_switches = []
+          networks = []
+          adapters = []
+          adapterCount = 2
           env[:machine].config.vm.networks.each do |type, opts|
             next if type != :public_network && type != :private_network
 
-            if opts[:bridge]
-              @logger.debug("Looking for switch with name or ID: #{opts[:bridge]}")
-              switch = switches.find{ |s|
-                s["Name"].downcase == opts[:bridge].to_s.downcase ||
-                  s["Id"].downcase == opts[:bridge].to_s.downcase
+            # Internal network is a special type which is shared between guests but not the host
+            if type == :private_network && opts[:intnet]
+              type = :internal_network
+            end
+
+            # Private network is the equivalent of virtualbox host-only network
+            # Annoyingly the private/internal terminology is flipped in hyper-v so an internal switch is what we want here
+            if type == :private_network
+              private_switch = managed_switches.find{ |s|
+                s["SwitchType"].downcase == "internal"
               }
-              if switch
-                @logger.debug("Found switch - Name: #{switch["Name"]} ID: #{switch["Id"]}")
-                switch = switch["Id"]
-                break
-              end
+              data = [:hostonly, opts]
+              additional_switches.append(private_switch["Id"])
             end
-          end
 
-          # If we already configured previously don't prompt for switch
-          sentinel = env[:machine].data_dir.join("action_configure")
+            # Again, the hyper-v terminology is flipped. We want a private switch for an internal network
+            if type == :internal_network
+              internal_switch = managed_switches.find{ |s|
+                s["SwitchType"].downcase == "private"
+              }
+              data = [:intnet, opts]
+              additional_switches.push(internal_switch["Id"])
+            end
 
-          if !switch && !sentinel.file?
-            if switches.length > 1
-              env[:ui].detail(I18n.t("vagrant_hyperv.choose_switch") + "\n ")
-              switches.each_index do |i|
-                switch = switches[i]
-                env[:ui].detail("#{i+1}) #{switch["Name"]}")
+            # Public networks will be bridged to a specified network or prompted if it can't be found
+            if type == :public_network
```

Comment:
This can easily be implemented with Hyper-V `External` switches but there is a limitation of only being able to bridge one virtual adapter to a physical adapter. At the moment the external switch would need to be made before running `vagrant up` but I think it would work (untested)